include/NovelRT.h

```diff
@@ -0,0 +1,112 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_H
+#define NOVELRT_H
+
+#include <math.h>
+#include <type_traits>
+#include <map>
+#include <string>
+#include <glad.h>
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_mixer.h>
+#include <SDL2/SDL_surface.h>
+#include <SDL2/SDL_image.h>
+#include <SDL2/SDL_events.h>
+#include <string>
+#include <map>
+#include <typeinfo>
+#include <vector>
+#include <functional>
+#include <algorithm>
+#include <memory>
+#include <glm/glm.hpp>
+#include <glm/gtx/rotate_vector.hpp>
+#include <stdint.h>
+#include <iostream>
+#include <stdexcept>
+#include <fstream>
+#include <sstream>
+#include <limits>
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include "spdlog/spdlog.h"
+#include "spdlog/sinks/stdout_color_sinks.h"
+#include "spdlog/async.h"
+
+namespace NovelRT {
+  typedef void (*NovelUpdateSubscriber)(double deltaSeconds);
+  typedef class DebugService DebugService;
+  typedef class LoggingService LoggingService;
+  typedef class NovelRunner NovelRunner;
+}
+
+namespace NovelRT::Audio {
+  typedef std::map<std::string, Mix_Chunk*> SoundBank;
```

Comment:
not for this pr, but it would be better to create an actual `SoundBank` type which is a simple wrapper

I'm pretty sure this will be needed for supporting sdl, xaudio, and openal

CC. @capnkenny

Reply:
(same for the other two below)

Reply:
Propagated to #102 . Resolving.